pandas/core/groupby/groupby.py

```diff
@@ -1519,7 +1519,11 @@ def _bool_agg(self, val_test, skipna):
 
         def objs_to_bool(vals: ArrayLike) -> tuple[np.ndarray, type]:
             if is_object_dtype(vals):
-                vals = np.array([bool(x) for x in vals])
+                # GH#37501: don't raise on pd.NA when skipna=True
+                if skipna:
```

Comment:
isn't this just `isna(values)`

Reply:
Think the `bool(x)` is needed here because we want to map values like 0 -> False and 1 -> True which `isna` won't support. The `else True` here is just arbitrary since the underlying algo masks missing values so that the value in that position won't matter.

Reply:
ok this is fine for a backport, but ideally we clean this up with a dedicated routine to do this (as i suspect we are doing this 'manual' checking elsewhere). can you create an issue / PR :->

Reply:
will do!